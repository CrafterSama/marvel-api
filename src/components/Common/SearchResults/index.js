import React, { useState } from 'react'
import { Link } from 'react-router-dom'
import { useStateValue } from '../../../context'
import CharacterCard from '../../Character/components/CharacterCard'
import CharacterModal from '../../Character/components/CharacterModal'
import Details from '../../Details/components/Details'
import {
  MainGrid,
  MainGridItem,
  Card,
  CardInfo,
  CardName,
  SectionTitle,
  Overlayer
} from '../../../styles'

const SearchResults = () => {

  const [state] = useStateValue()
  const [isOpen, setIsOpen] = useState(false)
  const [isFade, setIsFade] = useState(false)
  const [characterData, setCharacterData] = useState()

  const handleToggleModal = () => {
    setIsFade(!isFade)
    setIsOpen(!isOpen)
  }

  console.log(state)

  return (
    <>
      {state.characters &&
        <>
          <SectionTitle>Characters</SectionTitle>
          <MainGrid>
            {state.characters.map((item, key) => (
              <MainGridItem key={key}>
                <CharacterCard
                  data={item}
                  setData={setCharacterData}
                  showCharacterInfo={handleToggleModal}
                />
              </MainGridItem>
            ))}
          </MainGrid>
        </>
      }
      {state.comics &&
        <>
          <SectionTitle>Comics</SectionTitle>
          <MainGrid>
            {state.comics.map((item, key) => (
              <MainGridItem key={key}>
                <Card background={item.thumbnail}>
                  <Overlayer />
                  <Link to={`/comics/${item.id}`}>
                    <CardInfo>
                      <CardName>
                        <span>{item.title}</span>
                      </CardName>
                      <div className='favorite'></div>
                    </CardInfo>
                  </Link>
                </Card>
              </MainGridItem>
            ))}
          </MainGrid>
        </>
      }
      {state.comic &&
        <>
          <SectionTitle>Marvel Issue</SectionTitle>
          <MainGrid>
            {state.comic.map((item, key) => (
              <Details data={item} key={key} />
            ))}
          </MainGrid>
        </>
      }
      <CharacterModal
        isOpen={isOpen}
        data={characterData}
        toggle={handleToggleModal}
      />
    </>
  );
}

export default SearchResults
