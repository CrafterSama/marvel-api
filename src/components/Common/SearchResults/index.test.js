import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import SearchResults from './index';

configure({ adapter: new Adapter() });

describe('Test for <SearchResults /> Component', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<SearchResults data={{ data: {} }} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
