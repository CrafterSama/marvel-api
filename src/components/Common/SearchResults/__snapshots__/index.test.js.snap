// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <SearchResults /> Component matches snapshot 1`] = `
<Fragment>
  <CharacterModal
    isOpen={false}
    toggle={[Function]}
  />
</Fragment>
`;
